homeassistant/components/axis/device.py

```diff
@@ -95,32 +97,48 @@ def serial(self):
             self.hass.async_create_task(
                 self.hass.config_entries.async_forward_entry_setup(
                     self.config_entry, 'binary_sensor'))
+
+            self.api.stream.connection_status_callback = \
+                self.async_connection_status_callback
+            self.api.enable_events(event_callback=self.async_event_callback)
             self.api.start()
 
         return True
 
     @callback
-    def async_signal_callback(self, action, event):
+    def async_connection_status_callback(self, status):
+        """Handle signals of gateway connection status."""
+        from axis.streammanager import SIGNAL_PLAYING
+        if self.available != status == SIGNAL_PLAYING:
```

Comment:
This check looks weird.

Reply:
available is True if status is SIGNAL_PLAYING

Status will be reported every minute due to RTPS keep-alive, it will also trigger on other changes in status. The check is made to not do unnecessary updates on hass state machine

Reply:
I can add an extra description to the method

Reply:
I'm not sure this is working as you describe it:
```py
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.3.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: SIGNAL_PLAYING = 'playing'                                                                                                                                                                            

In [2]: status = 'playing'                                                                                                                                                                                    

In [3]: available = True                                                                                                                                                                                      

In [4]: check = available != status == SIGNAL_PLAYING                                                                                                                                                         

In [5]: check                                                                                                                                                                                                 
Out[5]: True

In [6]: available = False                                                                                                                                                                                     

In [7]: check = available != status == SIGNAL_PLAYING                                                                                                                                                         

In [8]: check                                                                                                                                                                                                 
Out[8]: True
```

Reply:
You're right, I realised this when I where doing the rebase. I've added parentheses for the status == part, just not pushed it yet

Reply:
I see you added a parenthesis. Good!